backend/engine/plugins/checkov/main.py

```diff
@@ -144,10 +205,12 @@ def parse_checkov(checkov_output: list[dict], repo_path: str, config_dir: str, c
     return (findings, error)
 
 
-def get_config_dir(repo_path: str, config: dict) -> tuple[str, str]:
+def get_config_dir(config: dict) -> tuple[Path, Optional[str]]:
     """
     Determine if config directory should be from S3 or default (local)
     """
+    # TODO: Write to the temporary volume instead of the plugin source directory.
```

Comment:
Nit: TODO possibly left in? Would this be better as an issue in an issue tracker? (also L222 below)

Reply:
Yep, this is intentional -- it's a note for the next PR in this series (or if I don't get around to it, will be added to the backlog).

The existing implementation of this writes the retrieved configuration into the actual plugin source directory, which persists between runs.  This *technically* works, so doesn't block this PR, but is definitely not good and potentially cause issues in some corner cases.

As far as I can tell, we're not actually using this "feature" so it's a lower priority to fix.